lib/block_res.rb

```diff
@@ -0,0 +1,14 @@
+module Stayappy
+  class BlockRes < Reservation
+    attr_reader :block, :discounted_rate
+  
+    def initialize(block, check_in, check_out, discounted_rate)
+     if block.count > 5 
+        raise ArgumentError.new("There can only be five rooms in a block.") 
+      end
+      @block = block
```

Comment:
consider giving this instance variable a name like `@rooms` to increase readability.